src/main/route/PromotionRouter.ts

```diff
@@ -24,6 +24,10 @@ export class PromotionRouter {
       '/:id/downVote',
       this.promotionController.downVotePromotion
     );
+    this.promotionRouter.get(
+      '/:id/restaurantDetails/',
+      this.promotionController.getRestaurantDetails
+    );
```

Comment:
Just confirming, this is the endpoint we'll call to get the restaurant details of a certain promotion?

In src/main/repository/PromotionRepository.ts, it looks like we get all the restaurant details along with the promotions so I was wondering if this was needed by some other part of the app? If not, I think it's fine to keep supporting this but I'm just wondering if we have a current use case on the FE

Reply:
Oh the one in promotionrepository does not contain the data back from google places api, that one contains just the Restaurant entity which is lat lon. To get the restaurant details itself, we need to make this separate request.

I'm not sure if we'll run into rate limiting issues if we get all promotions along with their restaurant details from places API. By having this separate, we only make this request when a user clicks on restaurant card

Reply:
Oh I see yeah that makes sense, thanks for explaining! Yeah this is a good solution cause we probably don't want to make those extra API calls if we don't need to